### ISSUE TITLE
Notebook doesn't open on recent Ubuntu (e.g. 20.04) with Chromium as default browser

### ISSUE THREAD

**Issue by mgedmin** [jupyter/notebook#5479]
Steps to reproduce:

```
$ jupyter-notebook
[I 14:59:35.624 NotebookApp] Serving notebooks from local directory: /home/mg
[I 14:59:35.624 NotebookApp] The Jupyter Notebook is running at:
[I 14:59:35.624 NotebookApp] http://localhost:8888/?token=a4a2d5dfeaf5a426507e887f3909cf787b3426e51a7d2954
[I 14:59:35.624 NotebookApp]  or http://127.0.0.1:8888/?token=a4a2d5dfeaf5a426507e887f3909cf787b3426e51a7d2954
[I 14:59:35.624 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 14:59:35.643 NotebookApp] 
    
    To access the notebook, open this file in a browser:
        file:///home/mg/.local/share/jupyter/runtime/nbserver-780941-open.html
    Or copy and paste one of these URLs:
        http://localhost:8888/?token=a4a2d5dfeaf5a426507e887f3909cf787b3426e51a7d2954
     or http://127.0.0.1:8888/?token=a4a2d5dfeaf5a426507e887f3909cf787b3426e51a7d2954
```

and then

![Ekrano nuotrauka iš 2020-05-25 15-00-31](https://user-images.githubusercontent.com/159967/82811669-fb359880-9e99-11ea-809d-7d798900739b.png)

This is because Chromium in Ubuntu 20.04 is packaged via [Snap](https://snapcraft.io/), which puts all applications in a container where they cannot see certain filesystem directories (anything outside /home and /media, basically, plus all the files in dotted directories are inacessible -- which cuts off ~/.local/share/).

Clicking the http://localhost:8888/... link works fine.

While checking if this issue was already filed I've discovered the [`use_redirect_file`](https://jupyter-notebook.readthedocs.io/en/stable/config.html#options) option, which should work as a workaround.

It is unfortunate that these two security features clash and make the user experience worse.  I wonder if anything can be done?

**Comment by dimpase** [comment]
Much of this could be avoided if `use_redirect_file` were `False` by default.

**Comment by mgedmin** [comment]
Incidentally, Ubuntu 22.04 LTS will make its the default browser, Firefox, a snap package as well, which presumably has the same sandbox restrictions.

**Comment by moorepants** [comment]
Yes, this is an issue on Ubuntu 22.04 with Firefox. Here is a recent stackoverflow post that demonstrates it: https://stackoverflow.com/questions/54877158/jupyter-notebook-wont-open-in-newly-installed-firefox-quantum

I just upgraded to 22.04 and `jupyter notebook` opens a tab in firefox with the error. I have to manually open one of the URLs.

**Comment by SirNate0** [comment]
Is there a way to get it working without manually entering the token? Dolphin (on KDE) will open a notebooks for me jupyter-notebook, but when I have `c.NotebookApp.use_redirect_file = True` I get this in Firefox
```txt
File not found

Firefox can’t find the file at /tmp/tmponpvtnj7.html.

    Check the file name for capitalization or other typing errors.
    Check to see if the file was moved, renamed or deleted.
```

If I have `c.NotebookApp.use_redirect_file = False` instead, then firefox opens the page successfully, but it is the login page and not the notebook page. E.g. it opens `http://localhost:8890/login?next=%2Fnotebooks%2FBezier-Fitting.ipynb` and asks me to enter the token. If I enter the token, then it does redirect to the notebook, but I have to manually copy and paste the token from the URL shown in the console.

If I manually enter the notebook URL with the token parameter (e.g. `http://localhost:8890/notebooks/Bezier-Fitting.ipynb?token=<COPIED_FROM_TERMINAL>`), then it will directly open the notebook page. Why is this not how jupyter actually opens the notebook? In fact, this is exactly the link that the redirect file is causing a refresh to.
